Devscord.DiscordFramework/Middlewares/Factories/UserRoleFactory.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System.Linq;
+using Devscord.DiscordFramework.Commons;
+using Devscord.DiscordFramework.Middlewares.Contexts;
+using Discord.Rest;
+using Discord.WebSocket;
+
+namespace Devscord.DiscordFramework.Middlewares.Factories
+{
+    public class UserRoleFactory
+    {
+        public UserRole Create(SocketRole socketRole)
+        {
+            var permissions = socketRole.Permissions.ToList().Select(x => (Permission) x).ToList();
```

Comment:
po co ToList()?

Reply:
w dodatku podwójne

Reply:
A to chyba dlatego, że wcześniej Permissions nie implementowało ICollection. Teraz można to usunąć.

Reply:
A jednak widzę, że tak musi być. Pierwsze ToList zwraca listę uprawnień, a nie obiekt Permissions, drugie ToList pozwala zamienić IEnumerable na ICollection